### PR TITLE
D8AGE-364: Drop openeuropa/code-review, use updated setup.

### PR DESCRIPTION
**Pull request by intelektron**
…resolved remaining errors.

### CODE REVIEW

File: grumphp.yml.dist
Comment (intelektron) [inline grumphp.yml.dist:2]: I copied all necessary things from here

File: src/ApiClient.php
Comment (intelektron) [inline src/ApiClient.php]: Now I noticed we are not adding types to properties, but it will be forced by higher levels of PHPStan anyway.

File: grumphp.yml.dist
Comment (intelektron) [inline grumphp.yml.dist]: I set the initial level to 5, it would be reasonable to raise it later and fix remaining issues

File: grumphp.yml.dist
Comment (drishu) [inline grumphp.yml.dist]: I think we should have a local ruleset in a resources folder similar to this https://github.com/openeuropa/code-review/blob/2.x/resources/library-phpcs-ruleset.xml but referencing /vendor/squizlabs/php_codesniffer/src/Standards/PSR12/ruleset.xml there, because I think it's good practice to harmonize the whitespace

File: grumphp.yml.dist
Comment (drishu) [reply grumphp.yml.dist]: lets bump it up to level 6 to catch the hint types

File: src/ApiClient.php
Comment (drishu) [reply src/ApiClient.php]: lets fix all issues reported after bumping up the level

**Review by drishu — CHANGES_REQUESTED** [review]
- Lets remove cweagans/composer-patches
- Since we added the token getter and setter lets cover them in a test in ClientTest

File: grumphp.yml.dist
Comment (intelektron) [inline grumphp.yml.dist:31]: Moved to config files, so it works with PHP Storm.
We probably don't need it as parameters, it is not set in any pipeline

File: composer.json
Comment (drishu) [inline composer.json:21]: don't forget to remove cweagans/composer-patches

File: phpcs.xml
Comment (drishu) [inline phpcs.xml]: let's use .dist with the naming so the file can be overwritten by a local setup for example, also add phpcs.xml to gitignored

File: phpstan.neon
Comment (drishu) [inline phpstan.neon]: let's use .dist with the naming so the file can be overwritten by a local setup for example, also add phpstan.neon to gitignored

File: phpcs.xml
Comment (drishu) [inline phpcs.xml]: intresting, does phpcs resolve the path itself to the one from vendor?

File: tests/src/Traits/ClientTestTrait.php
Comment (drishu) [inline tests/src/Traits/ClientTestTrait.php]: seems a bit excessive, I'd just do `array<mixed>` but its obviously not wrong, so up to you.

File: phpstan.neon
Comment (drishu) [inline phpstan.neon]: the docker-image has xdebug enabled on cli by default, so maybe we can add the xdebug option to disable it when it runs

**Review by drishu — CHANGES_REQUESTED** [review]

File: phpcs.xml
Comment (intelektron) [reply phpcs.xml]: Yes, if it is only added to PHPCS paths. The internal rules are there by default

File: tests/src/Traits/ClientTestTrait.php
Comment (intelektron) [reply tests/src/Traits/ClientTestTrait.php]: I added the long list when I was looking PHPStan level 9. Actually you are right, it is too verbose for level 8, and the usage of this array is strictly typed, so we get all the type hinting in PHPStorm.

I'll change it to `@var array<int, array<string, mixed>>`, to avoid PHPStorm throwing warnings that `array<mixed> is equivalent to array`,

File: phpstan.neon
Comment (intelektron) [reply phpstan.neon]: I am going to use the docker-compose's `-e XDEBUG_MODE=off` for the githook and pipeline.
However, I am not 100% happy with the solution, it will be difficult to apply it during local development

File: grumphp.yml.dist
Comment (intelektron) [inline grumphp.yml.dist:13]: I removed standard file paths from GrumPHP, they are detected automatically., If anybody adds custom `phpstan.neon` or `phpcs.xml`, it will work without changing the Grum's config

File: tests/src/ClientTest.php
Comment (intelektron) [inline tests/src/ClientTest.php:46]: Perhaps we should throw this dummy config into some trait constant, of default argument? At least for the cases when it's completely irrelevant

File: src/Endpoint/EndpointBase.php
Comment (intelektron) [inline src/Endpoint/EndpointBase.php:190]: This is added because of backwards compatibility. When we lower the dependencies, the `$request->withBody()` method changes the type of `$request` from `RequestInterface` to `MessageInterface`, while both interfaces are implemented in the class. Therefore, PHPStan thinks that `sendRequest($request);` gets the wrong parameter type. Apparently, the union types were not supported yet at the time.

The assert is neutral here, and it is respected by all the quality tools, and PHPStorm typehinting.

File: tests/src/ClientTest.php
Comment (drishu) [reply tests/src/ClientTest.php:46]: - I like the idea, but we have these all over,
ex 'mainApiEndpoint' we can find also in data providers, so if we add them we can't do them in bulk, we need for each config a separate key
ex `['mainApiEndpoint'=>'https://example.com/v2/CheckConnection']`
- trait constant are avail only starting php8.2
- maybe leave a todo, so when we get the second review we can re-discuss it

Comment (intelektron) [reply tests/src/ClientTest.php:46]: True, maybe let's wait for the rest of the endpoints, we'll see how it looks then

File: phpstan.neon
Comment (drishu) [reply phpstan.neon]: I dont lile it, the way to do it would be via XDEBUG_MODE in docker-compose yml, but that has other implications, so lets just drop the change for now

**Review by drishu — APPROVED** [review]

**Pull request merged**
